applications/teletype/src/teletype_fax_util.erl

```diff
@@ -22,129 +20,194 @@
 
 -spec add_data(kz_json:object()) -> kz_json:object().
 add_data(DataJObj) ->
+    IsPreview = teletype_util:is_preview(DataJObj),
     FaxBoxJObj = get_faxbox_doc(DataJObj),
     Values =
         props:filter_empty(
-          [{<<"error">>, error_data(DataJObj)}
+          [{<<"error">>, error_data(DataJObj, IsPreview)}
           ,{<<"faxbox">>, FaxBoxJObj}
           ,{<<"owner">>, get_owner_doc(DataJObj, FaxBoxJObj)}
-          ,{<<"fax">>, kz_doc:public_fields(maybe_get_fax_doc(DataJObj))}
+          ,{<<"fax">>, maybe_get_fax_doc(DataJObj, IsPreview)}
           ,{<<"timezone">>, find_timezone(DataJObj, FaxBoxJObj)}
           ]),
     kz_json:set_values(Values, DataJObj).
 
--spec find_timezone(api_ne_binary() | kz_json:object(), kz_json:object()) -> ne_binary().
-find_timezone('undefined', FaxBoxJObj) ->
-    kzd_fax_box:timezone(FaxBoxJObj);
-find_timezone(Timezone, _FaxBoxJObj) when is_binary(Timezone) ->
-    Timezone;
-find_timezone(DataJObj, FaxBoxJObj) ->
-    Paths = [<<"fax_timezone">>
-            ,[<<"fax_info">>, <<"fax_timezone">>]
-            ,[<<"fax">>, <<"tx_result">>, <<"timezone">>]
-            ,[<<"fax">>, <<"rx_result">>, <<"timezone">>]
-            ],
-    find_timezone(kz_json:get_first_defined(Paths, DataJObj), FaxBoxJObj).
+-spec add_attachments(kz_json:object(), kz_proplist(), boolean()) -> {kz_proplist(), attachments()}.
+add_attachments(DataJObj, Macros, ShouldTerminate) ->
+    IsPreview = teletype_util:is_preview(DataJObj),
+    FaxDoc = props:get_value(<<"fax">>, Macros),
+    case kz_json:is_json_object(FaxDoc)
+        andalso kz_json:is_empty(FaxDoc)
```

Comment:
@icehess it seems that if there is a nonempty FaxDoc it will not try to get the attachment ?